web/pf4/src/utils/events.ts

```diff
@@ -0,0 +1,81 @@
+interface EventItem {
+  legendName: string;
+  idx: number;
+  serieName: string;
+  onClick?: (props: any) => any;
+  onMouseOver?: (props: any) => any;
+}
+
+export type VCEvent = {
+  childName: string[];
+  target: string;
+  eventKey?: string;
+  eventHandlers: EventHandlers;
+};
+
+type EventHandlers = {
+  onClick?: () => EventMutation[],
+  onMouseOver?: () => EventMutation[],
+  onMouseOut?: () => EventMutation[]
+};
+
+type EventMutation = {
+  childName: string[];
+  target: string;
+  mutation: (props: any) => any;
+};
+
+export const addLegendEvent = (events: VCEvent[], item: EventItem): void => {
+  const eventHandlers: EventHandlers = {};
+  if (item.onClick) {
+    eventHandlers.onClick = () => {
+      return [
+        {
+          childName: [item.serieName],
+          target: 'data',
+          mutation: props => item.onClick!(props)
+        },
+        {
+          childName: [item.serieName],
+          target: 'data',
+          eventKey: 'all',
+          mutation: __ => null
+        }
+      ];
+    };
+  }
+  if (item.onMouseOver) {
+    eventHandlers.onMouseOver = () => {
+      return [
+        {
+          childName: [item.serieName],
+          target: 'data',
+          eventKey: 'all',
+          mutation: props => item.onMouseOver!(props)
+        }
+      ];
+    };
+    eventHandlers.onMouseOut = () => {
+      return [
+        {
+          childName: [item.serieName],
+          target: 'data',
+          eventKey: 'all',
+          mutation: __ => null
+        }
+      ];
+    };
+  }
+  events.push({
+    childName: [item.legendName],
+    target: 'data',
```

Comment:
Wasn't the original `target: ['data', 'labels']` better?
...or, perhaps, the original syntax was not working?

Reply:
Well, the original syntax did work but was undocumented and was the reason why we had some Proptypes warning in dev mode. I agree I would prefer the previous version if I could be sure it was "supported" and not just accidentally working. Quickly looking at victory's code, it was a mystery as too why it worked :-) and I don't get any response when asking there, so I prefer to stick to the documented path.